audio-analysis/lib/etl/sqlConnection.py

```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
     
     meta.create_all() # Create all tables
 
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `labelled` WHERE `url`="' + url + '" AND `'+ column_name +'` IS NOT NULL')
```

Comment:
I still don't understand the `column_name IS NOT NULL` part of the query. Why are we filtering by not null? If we import a video by default the pitch would be null so this method would return false

Reply:
Aaaaaa makes sense. @tjlam @Eldrick19 expained it to me. Basically it just checks if `column_name` has been filled in for a given `url` but the name of the method is a bit exotic.

Reply:
Why do we need to check if that column is filled though? If this returns false we don't want to update right?

Reply:
@tjlam You're right. I think it helps looking at this part of the code:
```
url_exists = urlExistsSpecific(url, engine, column_name)
if url_exists == True:
    print("This video is already in the database. Will overwrite ", column_name, " data.")
video.getAudio()
print('Getting Pitch...')
video.getPitchAnalysis()
print('Got Pitch\n')
print('Updating DB for Pitch...\n')
dfHelper.updateDBWithDataFromDF(video.pitch_list, engine, column_name)
print('Done Updating')
```

Reply:
Let's say that url is not in the database at all, would update still work?

Reply:
I assume it wouldn't??? because it would have to insert new rows

Reply:
That is correct, it assumes the URL is already in the DB